launcher/src/main/java/org/wso2/carbon/launcher/extensions/DropinsBundleDeployer.java

```diff
@@ -17,296 +17,46 @@
-                            if (!existingBundleInfoLine.getBundlePath().equals(newBundleInfoLine.getBundlePath())) {
-                                logger.log(Level.WARNING,
-                                        "Ignoring the deployment of bundle: " + newBundleInfoLine.toString() +
-                                                ", because it is already available in the system: " +
-                                                existingBundleInfoLine.getBundlePath() +
-                                                ". Bundle-SymbolicName and Bundle-Version headers are identical.");
-                                found = true;
-                                break;
-                            }
-                        } else {
-                            //  This means fragment-ness property is equal. Seems like we have a match.
-                            //  Now lets check whether their locations are equal. If the locations are equal, we don't
-                            //  need to add it again. But if the locations are different we should throw a WARN.
-                            if (existingBundleInfoLine.getBundlePath().equals(newBundleInfoLine.getBundlePath())) {
-                                //  As we have an exact match, no need to add again.
-                                logger.log(Level.FINE, "Deploying bundle: " + newBundleInfoLine.getBundlePath());
-                                found = true;
-                                break;
-
-                            } else {
-                                //  We have an exact match, but their locations are different.
-                                logger.log(Level.WARNING,
-                                        "Ignoring the deployment of bundle: " + newBundleInfoLine.toString() +
-                                                ", because it is already available in the system: " +
-                                                existingBundleInfoLine.getBundlePath() +
-                                                ". Bundle-SymbolicName and Bundle-Version headers are identical.");
-                                found = true;
-                                break;
-                            }
+                Optional<String> carbonProfile = Optional.ofNullable(System.getProperty("carbon.profile"));
+                if (carbonProfile.isPresent()) {
+                    DropinsBundleDeployerUtils.executeDropinsCapability(carbonHome.toString(), carbonProfile.get());
+                } else {
+                    DropinsBundleDeployerUtils.getCarbonProfiles().parallelStream().forEach(profileName -> {
+                        try {
+                            DropinsBundleDeployerUtils.executeDropinsCapability(carbonHome.toString(), profileName);
+                        } catch (IOException e) {
+                            logger.log(Level.SEVERE, "Failed to update the Carbon Profile : " + carbonProfile, e);
```

Comment:
Shouldn't we be revert everything and throw an exception since it failed to update the Carbon Profile? Why are we logging it and carrying forward?